src/common/Ulmus.hs

```diff
@@ -46,30 +46,23 @@ initializeApp :: (Show appState, Show action, Read appState, Read action, Typeab
 initializeApp (AppConfig update view updateDisplays initialAppState appPort setup) = do
     setup
     simpleWebApp appPort $ do
-        currentState <- local $ getState initialAppState
-        nextAction <- local (render $ view currentState)
-        currentState' <- local $ getState initialAppState
-        newState <- update nextAction currentState'
-        local (setState newState)
-        renderDisplay initialAppState updateDisplays
-
+        step view
+        loop (step updateDisplays)
+  where
+    step f = do
+      currentState <- local $ getState initialAppState
+      nextAction <- local (render $ f currentState)
+      currentState' <- local $ getState initialAppState
+      newState <- update nextAction currentState'
+      local (setState newState)
+    loop f = f >> loop f
```

Comment:
What's happening here? Could you explain (by commenting here, in the review) whats going on here? 😄